core/distributed/preconditioner/schwarz.cpp

```diff
@@ -47,30 +48,30 @@ Schwarz<ValueType, LocalIndexType, GlobalIndexType>::parse(
     const config::type_descriptor& td_for_child)
 {
     auto params = Schwarz::build();
-
-    if (auto& obj = config.get("generated_local_solver")) {
+    config::config_decorator decorator(config);
```

Comment:
decorator is not a particularly descriptive name, what if we named it `config` and renamed the parameter to something like `config_node`?

Reply:
we already used config for the object to hold property tree, so I would like to have the name easy to distinguish the behavior. 
`config_checker checker` is okay to me or `config_tracker tracker`?